src/lightning_app/components/training.py

```diff
@@ -0,0 +1,167 @@
+import logging
+import os
+from typing import List, Optional, Tuple, Type, Union
+
+from lightning import CloudCompute
+from lightning_app import LightningFlow, structures
+from lightning_app.components.python import TracerPythonScript
+from lightning_app.storage.path import Path
+
+_logger = logging.getLogger(__name__)
+
+
+class PyTorchLightningScriptRunner(TracerPythonScript):
+    def __init__(
+        self,
+        script_path: str,
+        script_args: Optional[Union[list, str]] = None,
+        node_rank: int = 1,
+        num_nodes: int = 1,
+        sanity_serving: bool = False,
+        cloud_compute: Optional[CloudCompute] = None,
+        **kwargs,
+    ):
+        super().__init__(
+            script_path, script_args, raise_exception=True, parallel=True, cloud_compute=cloud_compute, **kwargs
+        )
+        self.node_rank = node_rank
+        self.num_nodes = num_nodes
+        self.best_model_path = None
+        self.best_model_score = None
+        self.sanity_serving = sanity_serving
+        self.has_finished = False
+
+    def configure_tracer(self):
+        from pytorch_lightning import Trainer
+
+        tracer = super().configure_tracer()
+        tracer.add_traced(Trainer, "__init__", pre_fn=self._trainer_init_pre_middleware)
+        return tracer
+
+    def run(self, internal_urls: Optional[List[Tuple[str, str]]] = None):
+        if not internal_urls:
+            _logger.info(f"The node {self.node_rank} started !")
+            return
```

Comment:
Why can't `internal_urls` be required ?

Reply:
You need to wait for all work to be up. This is equivalent to doing a start.

Reply:
Would be great to add a comment on why we do it here.

Reply:
Done !